sklearn/notebooks/Online Prediction with scikit-learn.ipynb

```diff
@@ -207,37 +206,80 @@
     "    raw_training_data = pd.read_csv(train_data, header=None, names=COLUMNS)\n",
+    "\n",
     "# Remove the column we are trying to predict ('income-level') from our features list\n",
-    "train_features = raw_training_data.drop('income-level', axis=1)\n",
-    "# Create our training labels list\n",
-    "train_labels = (raw_training_data['income-level'] == ' >50K')\n",
+    "# Convert the Dataframe to a lists of lists\n",
+    "train_features = raw_training_data.drop('income-level', axis=1).as_matrix().tolist()\n",
+    "# Create our training labels list, convert the Dataframe to a lists of lists\n",
+    "train_labels = (raw_training_data['income-level'] == ' >50K').as_matrix().tolist()\n",
     "\n",
     "\n",
     "# Load the test census dataset\n",
     "with open('./census_data/adult.test', 'r') as test_data:\n",
     "    raw_testing_data = pd.read_csv(test_data, names=COLUMNS, skiprows=1)\n",
     "# Remove the column we are trying to predict ('income-level') from our features list\n",
-    "test_features = raw_testing_data.drop('income-level', axis=1)\n",
-    "# Create our training labels list\n",
-    "test_labels = (raw_testing_data['income-level'] == ' >50K.')\n",
-    "\n",
-    "\n",
-    "# Convert the categorical columns to a numerical value in both the training and testing dataset\n",
-    "encoders = {col:LabelEncoder() for col in CATEGORICAL_COLUMNS}\n",
-    "\n",
-    "for col in CATEGORICAL_COLUMNS:\n",
-    "    train_features[col] = encoders[col].fit_transform(train_features[col])\n",
-    "\n",
-    "for col in CATEGORICAL_COLUMNS:\n",
-    "    test_features[col] = encoders[col].fit_transform(test_features[col])\n",
+    "# Convert the Dataframe to a lists of lists\n",
+    "test_features = raw_testing_data.drop('income-level', axis=1).as_matrix().tolist()\n",
+    "# Create our training labels list, convert the Dataframe to a lists of lists\n",
+    "test_labels = (raw_testing_data['income-level'] == ' >50K.').as_matrix().tolist()\n",
+    "\n",
+    "\n",
+    "# Since the census data set has categorical features, we need to convert\n",
+    "# them to numerical values. We'll use a list of pipelines to convert each\n",
+    "# categorical column and then use FeatureUnion to combine them before calling\n",
+    "# the RandomForestClassifier.\n",
+    "categorical_pipelines = []\n",
+    "\n",
+    "for i, col in enumerate(COLUMNS[:-1]):\n",
+    "    if col in CATEGORICAL_COLUMNS:\n",
+    "        # Create a scores array to get the individual categorical column\n",
+    "        # Example:\n",
+    "        # data = [39, 'State-gov', 77516, 'Bachelors', 13, 'Never-married', 'Adm-clerical', \n",
+    "        #    'Not-in-family', 'White', 'Male', 2174, 0, 40, 'United-States']\n",
+    "        # scores = [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]\n",
+    "        #\n",
+    "        # Returns: [['Sate-gov']]\n",
+    "        scores = []\n",
+    "        for j, col2 in enumerate(COLUMNS[:-1]):\n",
```

Comment:
i, j, col, col2 are slightly confusing variable names

Reply:
I can switch that line to ` for j in range(len(COLUMNS[:-1])):`

i and j are indexes. 
col is the column

Reply:
actually removing col2 would make it clearer.

my confusion largely arose from *visualising* what the indexes are doing. the comments above the loops help but not immediately clear that we're creating the 1 hot array by simply comparing the column indexes.